src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -2290,14 +2290,15 @@ public void run()
             indexManager.truncateAllIndexesBlocking(truncatedAt);
             viewManager.truncateBlocking(replayAfter, truncatedAt);
 
-                SystemKeyspace.saveTruncationRecord(ColumnFamilyStore.this, truncatedAt, replayAfter);
-                logger.trace("cleaning out row cache");
-                invalidateCaches();
-
-            }
+            SystemKeyspace.saveTruncationRecord(ColumnFamilyStore.this, truncatedAt, replayAfter);
+            logger.trace("cleaning out row cache");
+            invalidateCaches();
         };
 
         runWithCompactionsDisabled(Executors.callable(truncateRunnable), true, true);
+
+        viewManager.build();
```

Comment:
I am a bit confused actually...did you mean build or start? From your comment it sounded more as you meant to call viewManager.start?

Reply:
> Can we also test the viewManager after truncate?

I don't understand this, the new `testTruncateWhileBuilding` waits for this second view build and verifies that it marks the view as built in the `system.built_views` table and that the MV is still empty. What else should we test? Maybe trying to write some data to the truncated view?

> I am a bit confused actually...did you mean build or start? From your comment it sounded more as you meant to call viewManager.start?

I don't see a `viewManager.start` method, do you mean renaming the method `TableViews#build()` to `TableViews#start()`? By the way, it's a bit confusing that `ColumnFamilyStore#viewManager` is a `TableViews` object, and not an actual `ViewManager`, I guess this is done so for the sake of symmetry with `ColumnFamilyStore#indexManager`.

Reply:
_I don't see a viewManager.start method, do you mean renaming the method TableViews#build() to TableViews#start()? By the way, it's a bit confusing that ColumnFamilyStore#viewManager is a TableViews object, and not an actual ViewManager, I guess this is done so for the sake of symmetry with ColumnFamilyStore#indexManager._

Yes, my bad It seems I switched again to the ViewManager class at some point. I. was thinking of changing the ViewManager name to reduce the confusion but not sure whether this will be appreciated now. 
I agree with all your points and double-checked the test. 
I think we are ready to commit this one, thank you!